src/backend/gl/src/device.rs

```diff
@@ -823,9 +831,17 @@ impl d::Device<B> for Device {
             {
                 let gl = &self.share.context;
                 gl.use_program(Some(name));
-                for (bname, binding) in name_binding_map.iter() {
-                    let loc = gl.get_uniform_location(name, bname);
-                    gl.uniform_1_i32(loc, *binding as _);
+                for (bname, (btype, binding)) in name_binding_map.iter() {
+                    match btype {
+                        n::BindingTypes::Images => {
+                            let loc = gl.get_uniform_location(name, bname);
+                            gl.uniform_1_i32(loc, *binding as _);
+                        }
+                        n::BindingTypes::UniformBuffers | n::BindingTypes::StorageBuffers => {
+                            let index = gl.get_uniform_block_index(name, bname).unwrap();
```

Comment:
I don't think `StorageBuffer` would end up as a uniform block in GLSL. We should limit this match arm to `UniformBuffers` only

Reply:
Yeah, I see.
But is it possible to retrieve storage block index by name?